lib/uinta/plug.ex

```diff
@@ -118,7 +126,11 @@ if Code.ensure_loaded?(Plug) do
       start = System.monotonic_time()
 
       Conn.register_before_send(conn, fn conn ->
-        if conn.request_path not in opts.ignored_paths || conn.status >= 300 do
+        # Log successful request if path is not filtered based on the sampling pool
+        # or log all HTTP status >= 300 (usually errors)
+        if (conn.request_path not in opts.ignored_paths &&
+              should_log?(opts[:success_log_sampling_ratio])) ||
```

Comment:
It feels like with this addition, the complexity of the clause becomes a lot higher. What would you think of moving all of this into a new `should_log_request?/2` that would take the `conn` and the `opts` and return a boolean that encompasses all of these conditions?